proto/palomachain/paloma/evm/tx.proto

```diff
@@ -39,3 +35,42 @@ message MsgRemoveSmartContractDeploymentRequest {
       [ (gogoproto.nullable) = false ];
 }
 message RemoveSmartContractDeploymentResponse {}
+
+message MsgUploadUserSmartContractRequest {
+  option (cosmos.msg.v1.signer) = "metadata";
+
+  palomachain.paloma.valset.MsgMetadata metadata = 1
+      [ (gogoproto.nullable) = false ];
+
+  string title = 2;
+  string abi_json = 3;
+  string bytecode = 4;
+  string constructor_input = 5;
+}
+
+message MsgUploadUserSmartContractResponse {
+  uint64 id = 1;
+}
+
+message MsgRemoveUserSmartContractRequest {
```

Comment:
The way I understand the code, a user is unable to interact (deploy/remove) with contracts that were not created by them, as they won't have access to their store. Just leaving this here for you to confirm.

Reply:
Yes, that's the idea. User only interact with their own contracts. :+1: